File: app/src/main/java/de/jbamberger/irremote/util/PanasonicTranslator.java
package de.jbamberger.irremote.util;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

class PanasonicTranslator extends CodeTranslator {

    PanasonicTranslator() {
        super(new int[]{3456, 1728}, new int[]{432}, new int[]{432, 432}, new int[]{432, 1296}, 37000);
    }

    @Override
    public int[] buildCode(String codeString) {
        return buildRawCode(hexToBytes(codeString));
    }
}

File: app/src/main/java/de/jbamberger/irremote/PanasonicTranslator.kt
package de.jbamberger.irremote

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

internal class PanasonicTranslator : CodeTranslator(
        initSequence = intArrayOf(3456, 1728),
        endSequence = intArrayOf(432),
        zero = intArrayOf(432, 432),
        one = intArrayOf(432, 1296),
        frequency = 37000) {

    override fun buildCode(codeString: String) = buildRawCode(hexToBytes(codeString))
}
